docs/src/dev/provider/index.asciidoc

```diff
@@ -54,6 +54,168 @@ This document attempts to address the needs of the different providers that have
 * Graph Language Provider
 * Graph Plugin Provider
 
+== HTTP API
```

Comment:
Something about the ordering of the Provider doc feels off to me. Someone implementing TinkerPop is the target reader. They likely would not start implementing by doing "HTTP API". The "Graph System Provider Requirements" are probably where most implementers would start their work. Could you give a bit of thought to organization and placement of this section?

I'd suggest a paragraph at the end of the "Provider Documentation" section above that says something like:

> Deciding which of the interfaces, protocols, and tests to implement depends largely on where your work fits in the TinkerPop system and how advanced you intend for that implementation to be. The following bullet points detail common provider projects that implementers undertake and what parts of the documentation will be most relevant to them in getting started:
> 
> * Building a TinkerPop compliant graph database 
> * Building a Gremlin Server implementation to provide remote connectivity over HTTP and with Gremlin drivers
> * Building a driver to work with Gremlin Server implementations
> * Building a Gremlin execution engine

The bullets are just possible sections that could likely replace the bullets above that just list the types of providers. Each could have some sort of introductory paragraph that gives an overview of what's involved in those tasks, links to docs we have and other hints to get started. I think that would introduce the sections below much better.

Reply:
I had it this way because I thought that the HTTP API was something both driver providers and graph providers should know, but I guess only driver providers need to directly implement it. I've pushed this whole section back into the graph driver providers section